src/main/java/org/testng/DataProviderInformation.java

```diff
@@ -0,0 +1,33 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+
+/**
+ * Represents information pertaining to a data provider that would be invoked.
+ */
+public class DataProviderInformation {
```

Comment:
Sure, but it's just a first step IMO :)
I'd love to have ITestNGMethod -> DataProviderMethod
                                                   -> TestMethod
But it won't be possible without breaking changes :(